run.go

```diff
@@ -67,20 +70,55 @@ func run(context *cli.Context) error {
 		return errors.New("invalid runtime config")
 	}
 
-	if err := create(context.Args().First(),
+	consolePath := context.String("console")
+
+	var wg sync.WaitGroup
+	var console Console
+	var err error
+	var consoleState *term.State
+
+	// if consolePath is /dev/ptmx or /dev/pts/ptmx
+	// means that we have to allocate a new pts
+	if consolePath == ptmxPath || consolePath == ptsPtmxPath {
+		console, err = newConsole()
+		if err != nil {
+			return err
+		}
+
+		// now consolePath is the slave pts
+		consolePath = console.Path()
+	}
+
+	if err = create(context.Args().First(),
 		context.String("bundle"),
-		context.String("console"),
+		consolePath,
 		context.String("pid-file"),
 		runtimeConfig); err != nil {
 		return err
 	}
 
+	detach := context.Bool("detach")
+
+	if !detach {
+		wg.Add(1)
+		go io.Copy(console, os.Stdin)
+		go func() {
+			defer wg.Done()
+			io.Copy(os.Stdout, console)
+		}()
+
+		// Save console state because it will be restored once container ends
+		consoleState, err = term.SetRawTerminal(os.Stdin.Fd())
```

Comment:
We're only saving stdin because:

1. we're modifying stdin (raw mode), so we need to restore it
2. there's no need to do anything on the internal pty the shim will use. If the shim needs to do somethig special to it, it'll do it (and in fact it puts that terminal in raw mode as well).
3. the lifetime of the pty created for the shim stop when `cc-runtime run` exits,